### PR TITLE
Allow COMS objects to be attached to emails

### PR DESCRIPTION
**Pull request by kyle1morel**
# Description

Allows for any documents that have been added to the submission to be selected in a modal and attached to the email.

baed69f Only send document ID for attachments. Back end controller unit tests.
1dba864 View and modal correctly tracking selected files
1cb56e3 Allow COMS objects to be attached to emails

[SHOWCASE-3596](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-3596)

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/src/components/file/FileSelectModal.vue [hunk 1]
```diff
@@ -0,0 +1,89 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import DocumentCard from '@/components/file/DocumentCard.vue';
```
Comment (jujaga) [inline frontend/src/components/file/FileSelectModal.vue:3]: Need whitespace line above this

File: frontend/src/components/file/FileSelectModal.vue [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+const emit = defineEmits(['fileSelect:submit']);
+
+// State
+const visible = defineModel<boolean>('visible');
+const selectedFiles: Ref<Array<Document>> = computed(() => props.selectedDocuments);
+
+// Actions
+function onDocumentClicked(data: { document: Document; selected: boolean }) {
+  if (data.selected) {
+    if (!selectedFiles.value.includes(data.document)) {
```
Comment (jujaga) [inline frontend/src/components/file/FileSelectModal.vue:29]: Might be able to collapse these conditionals together into one if statement if the converse else statements are fine being executed.

File: frontend/src/services/roadmapService.ts [hunk 3]
```diff
@@ -19,6 +19,9 @@ export default {
       emailData.bcc = parseCSV(emailData.bcc);
     }
 
-    return appAxios().put('roadmap', { activityId, emailData });
+    const attachmentIds = emailData.attachments;
+    delete emailData.attachments;
+
+    return appAxios().put('roadmap', { activityId, attachmentIds, emailData });
```
Comment (jujaga) [inline frontend/src/services/roadmapService.ts:25]: Something seems off about this - generally would be better to explicitly define the newly constructed object key and values than to have it be implicitly put together for safety/typing purposes. There might be a way of assigning this, without requiring the use of a delete statement.

**Review by wilwong89 — APPROVED** [review]

File: frontend/src/components/roadmap/Roadmap.vue [hunk 4]
```diff
@@ -175,4 +212,11 @@ onMounted(async () => {
       </div>
     </div>
   </Form>
+
+  <FileSelectModal
+    v-model:visible="fileSelectModalVisible"
+    :documents="documents"
+    :selected-documents="selectedFiles.slice()"
```
Comment (TimCsaky) [inline frontend/src/components/roadmap/Roadmap.vue:219]: testing in the browser.. when adding and removing attachments the result isnt consistent. maybe this should just be 
:selected-documents="selectedFiles"

**Review by jujaga — APPROVED** [review]

**Pull request merged**